molecule/provisioner/ansible/playbooks/docker/create.yml

```diff
@@ -20,7 +20,7 @@
 
     - name: Create Dockerfiles from image names
       template:
-        src: "{{ molecule_scenario_directory }}/Dockerfile.j2"
+        src: "{{ molecule_scenario_directory + '/' + (item.dockerfile | default( 'Dockerfile.j2')) }}"
```

Comment:
You'll need to update the `platforms_docker_schema` and add tests for it as well as the `test/resources/playbooks/docker/create.yml` to match this functionality under functional testing. Ideally, you could also document this in `molecule/driver/docker.py`.